aten/src/ATen/core/alias_info.h

```diff
@@ -117,3 +118,25 @@ inline std::ostream& operator<<(std::ostream& out, const AliasInfo& aliasInfo) {
   return out;
 }
 } // namespace c10
+
+namespace std {
+template <>
+  struct hash<c10::AliasInfo> {
+    size_t operator()(const c10::AliasInfo& aliasInfo) const {
+      auto hash = std::hash<bool>()(aliasInfo.isWrite());
+      for (auto &e: aliasInfo.beforeSets()) {
+        auto symbol_hash = std::hash<c10::Symbol>()(e);
+        hash = c10::hash_combine(hash, symbol_hash);
```

Comment:
This is not correct. Iteration order over unordered set is not specified. Two unordered sets can have the same contents but differing iteration order. There are a few ways to solve this problem but a cheap and cheerful one is to use a commutative operator like addition or xor to combine the hashes from the operations in the set.

Reply:
Got it, thanks! Let me try this out.